File: frontend/user/lib/data/mapper/category/category_mapper.dart
import 'package:b2205946_duonghuuluan_luanvan/data/models/category/category_model.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/entity/category/category.dart';

class CategoryMapper {
  static Category fromModel(CategoryModel model) {
    return Category(id: model.id, name: model.name);
  }

  static Map<String, dynamic> toJson(Category entity) {
    return {"id": entity.id, "name": entity.name};
  }
}

File: backend/app/infrastructure/database/mappers/category_mapper.py
from app.domain.entities.category_entity import CategoryEntity
from app.infrastructure.database.models.category import Category


class CategoryMapper:
    @staticmethod
    def to_entity(model: Category) -> CategoryEntity:
        return CategoryEntity(
            id=model.id,
            name=model.name,
            created_at=model.created_at,
        )
